src/Xhgui/Saver/File.php

```diff
@@ -1,17 +1,95 @@
 <?php
 
+/**
+ * File saving handler
+ */
 class Xhgui_Saver_File implements Xhgui_Saver_Interface
 {
-    private $_file;
+    /**
+     * @var string
+     */
+    private $file;
 
-    public function __construct($file)
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @var bool
+     */
+    private $separateMeta;
+
+    /**
+     * Xhgui_Saver_File constructor.
+     * @param string $path
+     * @param string $file or null for default
+     * @param bool $separateMeta
+     */
+    public function __construct($path, $file, $separateMeta = false)
     {
-        $this->_file = $file;
+        $this->path             = rtrim($path, '/\\').DIRECTORY_SEPARATOR;
+        $this->file             = !empty($file) ? $file : self::getFilename();
+        $this->separateMeta     = $separateMeta;
     }
 
+    /**
+     * @param array $data
+     * @return bool|int
+     */
     public function save(array $data)
     {
-        $json = json_encode($data);
-        return file_put_contents($this->_file, $json.PHP_EOL, FILE_APPEND);
+        if ($this->separateMeta) {
+            $profiles           = Xhgui_Util::getDataForStorage($data['profile'], true);
+
+            $meta               = $data['meta'];
+
+            // store summary in separate meta file to speed up aggregation
+            $meta['summary']    = $data['profile']['main()'];
+            $meta               = Xhgui_Util::getDataForStorage($meta, false);
+
+            file_put_contents($this->path.$this->file.'.meta',$meta.PHP_EOL, FILE_APPEND);
+
+            return file_put_contents($this->path.$this->file,$profiles.PHP_EOL, FILE_APPEND);
+        }
+
+        $json = Xhgui_Util::getDataForStorage($data);
+        return file_put_contents($this->path.$this->file, $json.PHP_EOL, FILE_APPEND);
+    }
+
+    /**
+     * Get filename to use to store data
+     * @param string $dir
+     * @return string
+     */
+    public static function getFilename() {
+
+        $fileNamePattern = '';
+
+        $prefix = 'xhgui.data.'.microtime(true);
+        
+        if (empty($_SERVER['REQUEST_URI'])) {
+            if (version_compare(PHP_VERSION, '7.0.0') >= 0) {
```

Comment:
use `PHP_VERSION_ID` constant instead of `version_compare` function

Reply:
What do  you mean? PHP_VERSION_ID is available as of PHP 5.2.7, version_compare is available event before that.

Reply:
5.3.0 is the version being supported: https://github.com/perftools/xhgui-collector/blob/1.7.0/composer.json#L17